pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -72,32 +67,47 @@ public class OfflineSegmentAssignment implements SegmentAssignment {
   private static final Logger LOGGER = LoggerFactory.getLogger(OfflineSegmentAssignment.class);
 
   private HelixManager _helixManager;
+  private TableConfig _tableConfig;
   private String _offlineTableName;
   private int _replication;
   private String _partitionColumn;
+  SegmentAssignmentStrategy _segmentAssignmentStrategy;
 
   @Override
   public void init(HelixManager helixManager, TableConfig tableConfig) {
     _helixManager = helixManager;
+    _tableConfig = tableConfig;
     _offlineTableName = tableConfig.getTableName();
     _replication = tableConfig.getValidationConfig().getReplicationNumber();
     ReplicaGroupStrategyConfig replicaGroupStrategyConfig =
         tableConfig.getValidationConfig().getReplicaGroupStrategyConfig();
     _partitionColumn = replicaGroupStrategyConfig != null ? replicaGroupStrategyConfig.getPartitionColumn() : null;
-
     if (_partitionColumn == null) {
       LOGGER.info("Initialized OfflineSegmentAssignment with replication: {} without partition column for table: {} ",
           _replication, _offlineTableName);
     } else {
       LOGGER.info("Initialized OfflineSegmentAssignment with replication: {} and partition column: {} for table: {}",
           _replication, _partitionColumn, _offlineTableName);
     }
+    //Check for dim table
+    if (_tableConfig.isDimTable()) {
```

Comment:
Can we handle dimTable within `getSegmentAssignmentStrategy()`?

Reply:
updated